### PR TITLE
Fix Vector::rotate()

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: vmmlib/vector.hpp [hunk 1]
```diff
@@ -248,9 +246,10 @@ class vector
     /** @return the product of all elements of this vector */
     T product() const;
 
+    /** Rotate this 3-component vector around the given axis. */
     template< typename TT >
-    vector< 3, T > rotate( const T theta, vector< M, TT > axis,
-                           typename enable_if< M == 3, TT >::type* = 0 ) const;
+    void rotate( T theta, vector< M, TT > axis,
```
Comment (tribal-tec) [inline vmmlib/vector.hpp]: Keep the const rotate, cf cross() ?

**Comment by eile** [comment]
Ping - do not merge, prep'ing downstream projects.

**Comment by tribal-tec** [comment]
free function rotate() missing?

**Comment by eile** [comment]
done

**Comment by tribal-tec** [comment]
+1

**Comment by eile** [comment]
...and more cleanups...

I'll start to push through the sub projects after this one.

**Comment by tribal-tec** [comment]
missing test for computeNearFar()

**Comment by eile** [comment]
I'm aware - will add this once I understand what it's supposed to do when doing the Livre changes.

**Pull request merged**